test/functional/http/post/1.second.secret.js

```diff
@@ -0,0 +1,438 @@
+					badTransactions.push(transaction);
+				});
+			});
+		});
+	});	
+
+	describe('confirmation', function () {
+
+		shared.confirmationPhase(goodTransactions, badTransactions);
+	});
+
+	describe('validation', function () {
+
+		it('using second passphrase on a fresh account should fail', function () {
+			transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, accountNoSecondPassword.password, accountNoSecondPassword.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.not.ok;
+				node.expect(res).to.have.property('message').to.equal('Sender does not have a second signature');
+				badTransactionsEnforcement.push(transaction);
+			});
+		});
+
+		describe('type 0 - sending funds', function () {
+
+			it('using no second passphrase should fail', function () {
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, account.password);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, account.password, 'invalid password');
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Failed to verify second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct second passphrase should be ok', function () {
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, account.password, account.secondPassword);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct empty second passphrase should be ok', function () {
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, accountEmptySecondPassword.password, accountEmptySecondPassword.secondPassword);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 1 - second secret', function () {
+
+			it('setting second signature twice on the same account should be not ok', function () {
```

Comment:
registering a second passphrase on an account with a second passphrase should fail